### PR TITLE
Add basic scala support using scalac

### PR DESCRIPTION
**Pull request by kalmiz**

### CODE REVIEW

File: ale_linters/scala/scalac.vim [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            continue
+        endif
+
+        let text = l:match[3]
+        let type = 'W'
+        if l:match[2] == 'error'
+            let type = 'E'
+        endif
+        let col = 0
+        if exists('a:lines[ln+1]')
```
Comment (w0rp) [inline ale_linters/scala/scalac.vim]: Can you not check `if ln + 1 < len(a:lines)` instead? It looks like you're doing a bounds check. If the string is empty for some reason, then `stridx` below will just return `-1` anyway.

File: ale_linters/scala/scalac.vim [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        if len(l:match) == 0
+            continue
+        endif
+
+        let text = l:match[3]
+        let type = 'W'
+        if l:match[2] == 'error'
+            let type = 'E'
+        endif
```
Comment (w0rp) [inline ale_linters/scala/scalac.vim]: You could have used a ternary expression for this, but this will do.

File: ale_linters/scala/scalac.vim [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        let col = 0
+        if exists('a:lines[ln+1]')
+            let col = stridx(a:lines[ln+1], '^')
```
Comment (w0rp) [inline ale_linters/scala/scalac.vim]: You should add spaces around the binary operator `+` here.

File: ale_linters/scala/scalac.vim (hunk 2 above)
Comment (kalmiz) [inline ale_linters/scala/scalac.vim]: I converted it to a ternary expression.

File: ale_linters/scala/scalac.vim (hunk 1 above)
Comment (kalmiz) [inline ale_linters/scala/scalac.vim]: You are right it's a bounds check. I rewrote it.

File: ale_linters/scala/scalac.vim (hunk 3 above)
Comment (kalmiz) [inline ale_linters/scala/scalac.vim]: I added the spaces.

**Comment by kalmiz** [comment]
Thanks for your comments! I have implemented your suggestions and added the documentation too.

**Pull request merged**